### ISSUE TITLE
Twitch Chat Message Event node not showing Interface reference

### ISSUE THREAD

**Issue by Shuisho10** [Yellow-Dog-Man/Resonite-Issues#2597]
### Describe the bug?

The Twitch Chat Message Event node (Network>Twitch>TwitchChatMessageEvent) appears to not show the bottom part where the reference for the Twitch Interface and the footer should be like the other nodes, making it hard to assign the interface reference.

### To Reproduce

1. Open a new session
2. Grab the protoflux tool and spawn the node browser.
3. Spawn the TwitchChatMessageEvent node (Network>Twitch>TwitchChatMessageEvent)
The node should appeared as showed in the screenshot.

### Expected behavior

The node should appear with the reference and the footer like the other nodes on Network>Twitch

### Screenshots

![image](https://github.com/user-attachments/assets/7c7b66c8-6611-4da6-8e84-ff1876c3d180)

### Resonite Version Number

 2024.7.15.1359

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[PC - 2024.7.15.1359 - 2024-07-17 01_03_12.log](https://github.com/user-attachments/files/16257418/PC.-.2024.7.15.1359.-.2024-07-17.01_03_12.log)

### Reporters

Shuisho

**Comment by shiftyscales** [comment]
Appending a screenshot depicting the missing interface @Shuisho10 is describing as seen on other Twitch related ProtoFlux nodes, in this case the Twitch Reward Redeem Event node.

![image](https://github.com/user-attachments/assets/7108b9dd-b083-4199-b257-f0888cffc7ae)

**Comment by AmasterAmaster** [comment]
This is also effecting the `Twitch Subscription Event` node as well.

I will also note the following weirdness with these nodes:
* Opening an inspector on these nodes will show that the fields are nonfunctional (fields are missing, and using a protoflux tool to attempt to take a reference of them will fail to spawn a node)
* Also while in the inspector, trying to click on the field button on the left of the broken fields will give a null context menu (just a center circle)
* Grabbing and pulling a wire (of the unique types like `Plan` and `CheerBadge`) and attempting to spawn a display node will fail.

**Comment by shiftyscales** [comment]
Thank you for the additional context and research into the issue, @AmasterAmaster. Is that a conclusive list? Is it just the two nodes you are aware of that have that issue, `Twitch Chat Message Event` and `Twitch Subscription Event`? Or are there others in that general category that also have this issue?

**Comment by AmasterAmaster** [comment]
@shiftyscales I just updated and I can confirm that these are the 2 nodes that are affected still as of build: `Beta 2024.7.25.1115`
This is currently the conclusive list for the twitch category of nodes, yes.

**Comment by Nytra** [comment]
`System.InvalidOperationException: Cannot assign non-data model type TwitchLib.Client.Enums.BadgeColor`

Should just need to mark `BadgeColor` as a data model type to fix this

**Comment by GrandtheUK** [comment]
I am also experiencing the same problem with the Twitch Chat Message Event node as @Shuisho10 on Beta 2024.7.25.1284

**Comment by o6uipi** [comment]
Has this been fixed yet.  It is now november and I am staring at the same problem posted in here in July.  I'm sure there are plenty of things to be working on, but since you all have spent so much money on advertising this should be considered something important since it allows Twitch people to show off a very useful and unique game feature to their audience

**Comment by ProbablePrime** [comment]
Thank you for your concern. Unfortunately, if there were any updates I'd let you know here. We're very busy with other items. Thank you!

**Comment by UnskilledWolf** [comment]
I found that the Twitch Interface field in the inspector of the Twitch Message Event node is functional. From what I can see only the UI is missing.
As a work around, it is possible to set the Interface on a working node, inspect that, and then drag the reference of the Interface from there to the Twitch Message Event node's component.
I have not tried packing or saving this, but it did work as is.

**Comment by o6uipi** [comment]
Can you describe how to do that I'm still new to scripting. I was also having problems getting it to actually see events from my channel so I don't know if I'm connecting it wrong

**Comment by UnskilledWolf** [comment]
Yes, I can make a more detailed description of how it works. I'm no expert either though, so there might be a better way of doing this.
To make the Twitch Interface work, you need to set the target user (For example using the User Inspector > Clicking your user > Dragging the "User" component into the User field). Second, you need to set the Channel to any *currently live* channel. I copied the name out of the Twitch URL of the channel.

For the workaround, you need to do the following:
1. Create a "Twitch Raid Event" and a "Twitch Chat Message Event"
2. Open both in the inspectors as shown (You may need to go up a slot, since <NODE_UI> was selected for me by default.) [A]
3. Drag the Twitch Interface into the slot of the "Twitch Raid Event" node. [B]
4. Drag the Twitch Interface reference from the "Twitch Raid Event" into the "Twitch Chat Message Event" [C]
5. To read the message, you need a setup like shown in my picture.

![Image](https://github.com/user-attachments/assets/fcfbc306-7d1b-4892-bd62-0f86c0bb6e3d)

If there's a cleaner way of doing this, I'd be curious to hear it too, but this works for me.

**Comment by Frooxius** [comment]
@ProbablePrime I can fix this very quickly (less than 5 minutes), do you mind if I take this from you?

**Comment by ProbablePrime** [comment]
@Frooxius Of course, go for it.

**Comment by o6uipi** [comment]
I got it working in a similar way UnskilledWolf, I ended up just linking the "twitch raid event" to the twitch interface, then after opening the inspector I dragged and dropped the reference from the raid event to the unassigned one on the message event inspector.  However to keep it working you have to save the twitch raid event node with the rest of it because if you delete it you'll lose your reference.  So I guess it's a jumper input
